### PR TITLE
DM-38398: Fix crashes exposed by rc2_subset.

### PR DESCRIPTION
**Pull request by erykoff**
This PR fixes a flag loading issue, and an index error that can come with very thin data like rc2_subset when all of the observations of a given star are flagged as bad due to having no flux after correcting for the local background.

### CODE REVIEW

File: python/lsst/fgcmcal/fgcmBuildFromIsolatedStars.py [hunk 1]
```diff
@@ -166,7 +166,7 @@ def setDefaults(self):
         source_selector = self.sourceSelector["science"]
         source_selector.setDefaults()
 
-        source_selector.doFlags = True
+        source_selector.doFlags = False
```
Comment (laurenam) [inline python/lsst/fgcmcal/fgcmBuildFromIsolatedStars.py:169]: Did you really mean to turn off (even the default science selector) flag filtering by default?

File: python/lsst/fgcmcal/fgcmBuildFromIsolatedStars.py [hunk 2]
```diff
@@ -388,8 +388,11 @@ def _make_all_star_obs_from_isolated_stars(
             self.config.apertureOuterInstFluxField + "Err",
         ]
 
+        local_background_flag_name = None
```
Comment (laurenam) [inline python/lsst/fgcmcal/fgcmBuildFromIsolatedStars.py:391]: I'm not seeing why this initialization is necessary (as the other use below is also behind a `if self.config.doSubtractLocalBackground:` clause)...

**Review by laurenam — APPROVED** [review]

File: python/lsst/fgcmcal/fgcmBuildFromIsolatedStars.py [hunk 3]
```diff
@@ -456,19 +456,33 @@ def _make_all_star_obs_from_isolated_stars(
...
-                for i, band in enumerate(self.config.requiredBands):
-                    (band_use,) = (sources[good_sources]["band"] == band).nonzero()
-                    np.add.at(
-                        n_req,
-                        (i, sources[good_sources]["obj_index"][band_use]),
-                        1,
-                    )
+                loop_bands = self.config.requiredBands
+            else:
+                loop_bands = np.unique(sources['band'])
```
[the remark this answers is not in the document]
Comment (erykoff) [reply python/lsst/fgcmcal/fgcmBuildFromIsolatedStars.py]: Argh!

File: python/lsst/fgcmcal/fgcmBuildFromIsolatedStars.py (hunk 1 above)
Comment (erykoff) [reply python/lsst/fgcmcal/fgcmBuildFromIsolatedStars.py:169]: Yes, this is intentional.  When running from the isolated stars the flags have already all been applied in the original isolated star association task, except for the local background flag.  What I have now may be redundant I realize, but it does work.

**Pull request merged**